components/CommonCompHabitChart.js

```diff
@@ -27,20 +28,61 @@ const gridRow = (row, clr) => {
 };
 
 const mapUserData = (users, clr) => (users.map(user => (
-        gridRow( user['recent'], clr)
+        gridRow( user, clr)
     )));
 
 const mapUserNames = users => (users.map(user => (
-    <View style={styles.textContainer}>
+    <View style={styles.textContainer} key={user}>
       <Text style={styles.paragraph}>
-      {user['name']}
+      {user}
       </Text>
     </View>
 )));
 
-const CommonCompHabitChart = ({ groupMembersData, groupColor }) => {
+const CommonCompHabitChart = ({ groupMembersData, groupMembersNames, groupColor }) => {
+    const nameList = groupMembersNames;
+    const dataList = [];
+    const startingDates = [];
+    for (var key in groupMembersData) {
+        if (groupMembersData.hasOwnProperty(key)) {     
+            //nameList.push(key);
 
-  //const recentHabit = [{'name': 'test0', 'recent': [1,0,1,0,0,0,1,1,0,0,1,1,0,0,1,1,1,0,1,1,1,0]}, {'name': 'test1', 'recent': [1,0,0,1,1,0,1,0,1,1,0,0,0,1,0,1,0,1,1,   1,0,0]}, {'name': 'test2', 'recent': [1,1,1,1,0,1,0,1,0,0,1,1,1,0,0,1,1,0,0,1,1,1]}];
+            startingDates.push(Math.min(...Object.keys(groupMembersData[key])));
+            dataList.push(groupMembersData[key]);
+        }
+    }
+    const moment = require('moment')
+
+    //This starts the habit chart at the earliest date any of the users completed the habit
+    let firstDay = ''+Math.min(...startingDates);
+    firstDay = (new Date(firstDay.substring(0,4), parseInt(firstDay.substring(4,6))-1, firstDay.substring(6)));
```

Comment:
nit: maybe add a comment `// attempt to get the earliest day in YYYYMMDD` for better clarity.